### ISSUE TITLE
Corrupt files because HTTP contentType and real file type do not match

### ISSUE THREAD

**Issue by SergioEstevao** [Automattic/pocket-casts-ios#1933]
This zendesk tickets reports that episode of the following podcasts are corrupted when downloaded:

- https://play.pocketcasts.com/podcasts/f77f2320-60c8-0139-33f3-0acc26574db2
- https://play.pocketcasts.com/podcasts/e8a4ad80-09e9-013c-f5ae-0acc26574db2

I debugged the issue and found out the following:
- When downloading the episode files the server is returning contentType as `video/mp4`
- The actual file being downloaded is an `audio/mp3` file
- Because we use the [contentType to check for the real file type](https://github.com/Automattic/pocket-casts-ios/blob/05d568c55ca40c3065be06f75c4890c76310b3f9/Modules/Server/Sources/PocketCastsServer/Private/API%20Tasks/MetadataTask.swift#L51) we then use that type to check if it's a video file or not and also set the extension of the local file using that contentType. This causes the system to fail when opening file.

In order to fix this we will need to change our code to actual look at the mime type written inside the file.

One possibility is to do something like this after we donwload the file:

```
import Foundation

func mimeType(forFileAtPath path: String) -> String {
    let fileExtension = (path as NSString).pathExtension
    // the code bellow should be changed to only read the bytes we need from the file instead of loading everything into memory
    guard let fileData = try? Data(contentsOf: URL(fileURLWithPath: path)) else {
        return "application/octet-stream"
    }
    
    let bytes = [UInt8](fileData)
    
    var fileHeader = [UInt8](repeating: 0, count: 12)
    for i in 0..<min(fileHeader.count, bytes.count) {
        fileHeader[i] = bytes[i]
    }
    
    let headerString = fileHeader.map { String(format: "%02x", $0) }.joined()
    
    let mimeTypes: [String: String] = [
        "25504446": "application/pdf",
        "504b0304": "application/zip",
        "47494638": "image/gif",
        "ffd8ff": "image/jpeg",
        "89504e47": "image/png"
        // Add more signatures as needed
    ]
    
    for (signature, mimeType) in mimeTypes {
        if headerString.hasPrefix(signature) {
            return mimeType
        }
    }
    
    return "application/octet-stream"
}

```

**Comment by igotdes** [comment]
Also reported here: https://forums.pocketcasts.com/forums/topic/bug-downloaded-video-podcast-episodes-are-not-playable-reported-as-corrupted/

The episodes download as MP4 files, but the content type is returning `binary/octet-stream`.

**Comment by joashrajin** [comment]
More reports here: 8822257-zd-a8c

**Comment by joashrajin** [comment]
8851625-zd-a8c

**Comment by SergioEstevao** [comment]
Solved by #2182
